test/e2e/suite/issuers/acme/certificate/http01.go

```diff
@@ -277,6 +279,77 @@ var _ = framework.CertManagerDescribe("ACME Certificate (HTTP01)", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("should allow updating the dns name of a failing certificate that had a wrong dns name", func() {
+		certClient := f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name)
+
+		By("Creating a failing Certificate")
+		cert := gen.Certificate(certificateName,
+			gen.SetCertificateSecretName(certificateSecretName),
+			gen.SetCertificateIssuer(cmmeta.ObjectReference{Name: issuerName}),
+			gen.SetCertificateDNSNames("google.com"),
+		)
+		cert.Namespace = f.Namespace.Name
+
+		_, err := certClient.Create(context.TODO(), cert, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Making sure the Order failed with a 400 since google.com is invalid")
+		order := &cmacme.Order{}
+		err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
+			orders, err := listOwnedOrders(f.CertManagerClientSet, cert)
+			Expect(err).NotTo(HaveOccurred())
+
+			if len(orders) != 1 {
+				log.Logf("Waiting as one Order should exist, but we found %d", len(orders))
+				return false, nil
+			}
+			order = orders[0]
+
+			expected := `400 urn:ietf:params:acme:error:rejectedIdentifier: Cannot issue for "google.com"`
```

Comment:
That's one thing I always do on purpose for strings that are meant for testing for two reasons:

1. To make sure that we "notice" that this behavior changed: if the "gotten" string changes, then someone will have to change the above "expected" string intentionally. If I was to use const variables, we wouldn't be able to notice behavior changes from the test cases.
2. Also because test cases must be crystal clear and offuscating things with const variables makes it hard to peruse.

(that's my opinion of course 😇)